main.tf

```
@@ -15,7 +15,7 @@ resource "google_project_iam_custom_role" "castai_role" {
   role_id     = local.custom_role_id
   title       = "Role to manage GKE cluster via CAST AI"
   description = "Role to manage GKE cluster via CAST AI"
-  permissions = toset(data.castai_gke_user_policies.gke.policy)
+  permissions = length(var.castai_role_permissions) > 0 ? var.castai_role_permissions : toset(data.castai_gke_user_policies.gke.policy)
```

How about making these permission from the variable additional permissions? This way the user has only to provide the additional permission and not all of them.
```suggestion
  permissions = setunion(toset(data.castai_gke_user_policies.gke.policy), toset(var.castai_role_permissions))
```